### ISSUE TITLE
[Muxer] Add sanity checks when computing mappings

### ISSUE THREAD

**Issue by iwahbe** [pulumi/pulumi-terraform-bridge#949]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

**Comment by t0yv0** [comment]
Specifically resources and functions should be disjoint in a muxed provider. For every particular resource or function, there should be no question which underlying provider will serve it. If there's an overlap this should warn the user loud and clear as early in the lifecycle as possible (tfgen time?) and not defer to runtime issues.

**Comment by iwahbe** [comment]
> Specifically resources and functions should be disjoint in a muxed provider. For every particular resource or function, there should be no question which underlying provider will serve it. If there's an overlap this should warn the user loud and clear as early in the lifecycle as possible (tfgen time?) and not defer to runtime issues.

I disagree. We don't need resources and functions to be disjoint, and warning in that case is unnecessary for our common use case: muxing a sdk and pf based provider. All resources in the pf based provider will be present in the sdk based provider. We will use the first one we come across.

The sanity check need only apply to types that are referenced by resources/functions in multiple schemas.

**Comment by t0yv0** [comment]
Care to elaborate. We just established experimentally that at Terraform level say AWS a resource is present either in the PF provider or SDKv2 provider not both. Any example they're present in both? If they're present in both the dispatch needs to - what - make a positional choice to chose leftmost? That sounds suspect.

**Comment by iwahbe** [comment]
There has been interest in using the muxer to override resources defined in bridged providers with native versions.
